servicetalk-transport-netty-internal/src/testFixtures/java/io/servicetalk/transport/netty/internal/AddressUtils.java

```diff
@@ -56,7 +57,17 @@ public static InetSocketAddress localAddress(final int port) {
      * @return a {@link HostAndPort} representation of server's listening address.
      */
     public static HostAndPort serverHostAndPort(final ServerContext ctx) {
-        return HostAndPort.of((InetSocketAddress) ctx.listenAddress());
+        return serverHostAndPort(ctx.listenAddress());
+    }
+
+    /**
+     * Returns a {@link HostAndPort} representation of server's listening address.
+     *
+     * @param address The {@link SocketAddress} of the server
+     * @return a {@link HostAndPort} representation of server's listening address.
+     */
+    public static HostAndPort serverHostAndPort(final SocketAddress address) {
```

Comment:
Since it would otherwise fail at runtime, consider using InetSocketAddress directly as the type for the argument

Reply:
The goal was to side cast to `InetSocketAddress` too.
Since it's a test utility only, expectation is that if there is a class cast issue, it will blow up for users immediately. I will keep it as-is for now. Thanks for noticing!